pkg/virt-controller/watch/vm.go

```diff
@@ -619,6 +620,43 @@ func (c *VMController) generateVMIMemoryDumpVolumePatch(vmi *virtv1.VirtualMachi
 	return err
 }
 
+func needUpdatePVCMemoryDumpAnnotation(pvc *k8score.PersistentVolumeClaim, request *virtv1.VirtualMachineMemoryDumpRequest) bool {
+	if pvc.GetAnnotations() == nil {
+		return true
+	}
+	annotation, hasAnnotation := pvc.Annotations[virtv1.PVCMemoryDumpAnnotation]
+	return !hasAnnotation || (request.Phase == virtv1.MemoryDumpUnmounting && annotation != *request.FileName) || (request.Phase == virtv1.MemoryDumpFailed && annotation != failedMemoryDump)
+}
+
+func (c *VMController) updatePVCMemoryDumpAnnotation(vm *virtv1.VirtualMachine) error {
+	request := vm.Status.MemoryDumpRequest
+	pvc, err := c.getPersistentVolumeClaimFromCache(vm.Namespace, request.ClaimName)
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error getting PersistentVolumeClaim to update memory dump annotation: %v", err)
+		return err
+	}
+	if pvc == nil {
+		log.Log.Object(vm).Errorf("Error getting PersistentVolumeClaim to update memory dump annotation: %v", err)
+		return fmt.Errorf("Error when trying to update memory dump annotation, pvc %s not found", request.ClaimName)
+	}
+
+	if needUpdatePVCMemoryDumpAnnotation(pvc, request) {
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string))
+		}
+		if request.Phase == virtv1.MemoryDumpUnmounting {
+			pvc.Annotations[virtv1.PVCMemoryDumpAnnotation] = *request.FileName
```

Comment:
Would a full path, relative to root of PVC be better?

Reply:
what do you mean by full path? isnt the path relative to the pod that mounts the pvc? like if I do a consumer pod with /dev/fs-pvc the path will be /dev/fs-pvc/<file-name> but if it is just under fs-pvc then the path is different

Reply:
>  `/memdump` is not much better than `memdump`

didnt quite understand the last part, but joining the pod mount dir with this annotation is good for the cp from the pod.